06-job-control.md

```diff
@@ -4,8 +4,200 @@ title: Extra Unix Shell Material
 subtitle: Job control
 minutes: 5
 ---
+
 > ## Learning Objectives {.objectives}
 >
 > * FIX ME
 
-FIX ME
\ No newline at end of file
+Our next topic is how to control programs *once they're running*. This
+is called [job control](glossary.html#job-control).  While it's less
+important today than it was back in the Dark Ages, it is coming back
+into its own as more people begin to leverage the power of computer
+networks, as job control allows you to run multiple processes from a single 
+bash session on a remote server.  Job control is also helpful for 
+running programs on your local computer and trying to deal with run-amok 
+scripts.  
+
+When we talk about controlling programs, what we really mean is
+controlling *processes*. A process is a program
+that's in memory and executing. Some of the processes on your computer
+are yours: they're running programs you explicitly asked for, like your
+web browser. Many others belong to the operating system that manages
+your computer for you, or, if you're on a shared machine, to other
+users. You can use the `ps` command to list them, just as you use `ls`
+to list files and directories:
+
+~~~{.input}
+$ ps
+~~~
+~~~{.output}
+PID    PPID   PGID   TTY   UID     STIME   COMMAND
+2152      1   2152   con  1000  13:19:07   /usr/bin/bash
+2276   2152   2276   con  1000  14:53:48   /usr/bin/ps
+~~~
+
+Every process has a unique process id (PID). Remember, this is a
+property of the process, not of the program that process is executing:
+if you are running three instances of your browser at once, each will
+have its own process ID.  However, it is worth noting that your computer 
+has a limited stock of PIDs, so over the long run, processes will reuse
+PIDs as they become free.  
+
+The second column in this listing, PPID, shows the ID of each process's
+parent. Every process on a computer is spawned by another, which is its
+parent (except, of course, for the bootstrap process that runs
+automatically when the computer starts up).
+
+The third column (labelled PGID) is the ID of the *process group* this
+process belongs to. We won't discuss process groups in this lecture, but
+they're often used to manage sets of related processes. Column 4 shows
+the ID of the terminal this process is running in. Once upon a time,
+this really would have been a terminal connected to a central timeshared
+computer. It isn't as important these days, except that if a process is
```

Comment:
Could slot in a sentence like

> On modern unix systems this terminal is *virtual* and the process (normally) gets its input from and sends its output and error messages to this terminal.

The following sentence can then be shortened to

> If a process is ...